tests/functional/api/test_boards.py

```diff
@@ -1,11 +1,18 @@
+from gitlab import GitlabDeleteError
+
+
 def test_project_boards(project):
     assert not project.boards.list()
 
     board = project.boards.create({"name": "testboard"})
     board = project.boards.get(board.id)
 
-    project.boards.delete(board.id)
-    assert not project.boards.list()
+    # Raises:
+    #   GitlabDeleteError: If the server cannot perform the request (the return code is not 2xx)
+    try:
+        project.boards.delete(board.id)
+    except GitlabDeleteError as error:
+        assert False, f"project.boards.delete raise error {error}"
```

Comment:
@TimKnight-DWP I wonder if we really even need to wrap and re-raise this. The traceback we get from exposing the error might be more helpful than just a failing assert, since the error would be unexpected. WDYT? Then we just simplify and call the delete method.

This would apply to most other deletes as well, and the diff here would be a lot smaller too and it'll be easier for me to review. Sorry for the back and forth :bow:

Reply:
@nejch 🤔 if we did that it'd come up as an "Error" rather than a "Fail" 

I'm happy to go through and do that